core/UrlHelper.php

```diff
@@ -155,6 +155,14 @@ public static function getArrayFromQueryString($urlQuery)
         if (strlen($urlQuery) == 0) {
             return array();
         }
+
```

Comment:
Instead of adding caching into this Url parsing method it would make more sense to cache at the caller level if possible. This method is just parsing a Url, it shouldn't be aware of any cache.